File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/mix/MixTraceApi.java
package com.bitget.openapi.api.mix;

import com.bitget.openapi.dto.request.mix.MixCloseTrackOrderReq;
import com.bitget.openapi.dto.request.mix.MixTraceModifyTPSLOrderReq;
import com.bitget.openapi.dto.request.mix.MixTraceSetCopyTradeSymbolReq;
import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * @Author: bitget-sdk-team
 * @Date: 2022-09-30 10:46
 * @DES: mix trace api
 */
public interface MixTraceApi {

    /**
     * Dealer closing interface
     * @param mixCloseTrackOrderReq
     * @return ResponseResult
     */
    @POST("/api/mix/v1/trace/closeTrackOrder")
    Call<ResponseResult> closeTraceOrder(@Body MixCloseTrackOrderReq mixCloseTrackOrderReq);

    /**
     * The trader obtains the current order
     * @param symbol
     * @param productType
     * @param pageSize
     * @param pageNo
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/currentTrack")
    Call<ResponseResult> currentTrack(@Query("symbol") String symbol,
                                      @Query("productType") String productType,
                                      @Query("pageSize") int pageSize,
                                      @Query("pageNo") int pageNo);

    /**
     * The trader obtains the historical order
     * @param startTime
     * @param endTime
     * @param pageSize
     * @param pageNo
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/historyTrack")
    Call<ResponseResult> historyTrack(@Query("startTime") String startTime,
                                      @Query("endTime") String endTime,
                                      @Query("pageSize") int pageSize,
                                      @Query("pageNo") int pageNo);

    /**
     * Summary of traders' profit sharing
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/summary")
    Call<ResponseResult> summary();

    /**
     * Historical profit sharing summary of traders (by settlement currency)
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/profitSettleTokenIdGroup")
    Call<ResponseResult> profitSettleTokenIdGroup();

    /**
     * Historical profit sharing summary of traders (by settlement currency and date)
     * @param pageSize
     * @param pageNo
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/profitDateGroupList")
    Call<ResponseResult> profitDateGroupList(@Query("pageSize") int pageSize,
                                             @Query("pageNo") int pageNo);


    /**
     * Historical profit distribution details of traders
     * @param marginCoin
     * @param date
     * @param pageSize
     * @param pageNo
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/profitDateList")
    Call<ResponseResult> profitDateList(@Query("marginCoin") String marginCoin,
                                        @Query("date") String date,
                                        @Query("pageSize") int pageSize,
                                        @Query("pageNo") int pageNo);

    /**
     * Details of traders to be distributed
     * @param pageSize
     * @param pageNo
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/waitProfitDateList")
    Call<ResponseResult> waitProfitDateList(@Query("pageSize") int pageSize,
                                             @Query("pageNo") int pageNo);

    /**
     * Followers obtain documentary information
     * @param pageSize
     * @param pageNo
     * @param startTime
     * @param endTime
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/followerHistoryOrders")
    Call<ResponseResult> followerHistoryOrders(@Query("pageSize") String pageSize,
                                            @Query("pageNo") String pageNo,
                                            @Query("startTime") String startTime,
                                            @Query("endTime") String endTime);

    /**
     * trader get copytrade symbol
     * @return ResponseResult
     */
    @GET("/api/mix/v1/trace/traderSymbols")
    Call<ResponseResult> traderSymbols();

    /**
     * trader set copytrade symbol
     * @return ResponseResult
     */
    @POST("/api/mix/v1/trace/setUpCopySymbols")
    Call<ResponseResult> setUpCopySymbols(@Body MixTraceSetCopyTradeSymbolReq mixTraceSetCopyTradeSymbol);

    /**
     * trader modify tpsl order
     * @return ResponseResult
     */
    @POST("/api/mix/v1/trace/modifyTPSL")
    Call<ResponseResult> modifyTPSL(@Body MixTraceModifyTPSLOrderReq mixTraceModifyTPSLOrderReq);

}

File: bitget-php-sdk-api/src/api/mix/MixTraceApi.php
<?php


namespace bitget\api\mix;


use bitget\internal\BitgetApiClient;
use bitget\model\mix\trace\CloseTrackOrderReq;

class MixTraceApi
{
    const BASE_URL = "/api/mix/v1/trace";

    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }
    /**
     * Dealer closing interface
     * @param $closeTrackOrderReq
     * @return string
     */
    public function closeTraceOrder(CloseTrackOrderReq $closeTrackOrderReq):string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/closeTrackOrder", $closeTrackOrderReq);
    }
    /**
     * The trader obtains the current order
     * @param $symbol
     * @param $productType
     * @param $pageSize
     * @param $pageNo
     * @return string
     */
    public function currentTrack(string $symbol,string $productType,string $pageSize,string $pageNo):string
    {
        $params = array("symbol" => $symbol, "productType" => $productType, "pageSize" => $pageSize, "pageNo" => $pageNo);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/currentTrack", $params);
    }

    /**
     * The trader obtains the historical order
     * @param $startTime
     * @param $endTime
     * @param $pageSize
     * @param $pageNo
     * @return string
     */
    public function historyTrack(string $startTime,string $endTime,string $pageSize,string $pageNo):string
    {
        $params = array("startTime" => $startTime, "endTime" => $endTime, "pageSize" => $pageSize, "pageNo" => $pageNo);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/historyTrack", $params);
    }
    /**
     * Summary of traders' profit sharing
     * @return string
     */
    public function summary():string
    {
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/summary", null);
    }
    /**
     * Historical profit sharing summary of traders (by settlement currency)
     * @return string
     */
    public function profitSettleTokenIdGroup():string
    {
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/profitSettleTokenIdGroup", null);
    }
    /**
     * Historical profit sharing summary of traders (by settlement currency and date)
     * @param $pageSize
     * @param $pageNo
     * @return string
     */
    public function profitDateGroupList(string $pageSize,string $pageNo):string
    {
        $params = array("pageSize" => $pageSize, "pageNo" => $pageNo);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/profitDateGroupList", $params);
    }
    /**
     * Historical profit distribution details of traders
     * @param $marginCoin
     * @param $date
     * @param $pageSize
     * @param $pageNo
     * @return string
     */
    public function profitDateList(string $marginCoin,string $date,string $pageSize,string $pageNo):string
    {
        $params = array("marginCoin" => $marginCoin, "date" => $date, "pageSize" => $pageSize, "pageNo" => $pageNo);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/profitDateList", $params);
    }
    /**
     * Details of traders to be distributed
     * @param $pageSize
     * @param $pageNo
     * @return string
     */
    public function waitProfitDateList(string $pageSize,string $pageNo):string
    {
        $params = array("pageSize" => $pageSize, "pageNo" => $pageNo);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/waitProfitDateList", $params);
    }

    /**
     * Followers obtain documentary information
     * @param $pageSize
     * @param $pageNo
     * @param $startTime
     * @param $endTime
     * @return string
     */
    public function followerHistoryOrders(string $pageSize,string $pageNo,string $startTime,string $endTime):string
    {
        $params = array("pageSize" => $pageSize, "pageNo" => $pageNo,"startTime"=>$startTime,"endTime"=>$endTime);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/followerHistoryOrders", $params);
    }
}